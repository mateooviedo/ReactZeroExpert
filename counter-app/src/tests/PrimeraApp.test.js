import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
// import { render } from '@testing-library/react';

import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraAPP />', () => {
	// test('Demostrar el mensaje hola soy goku', () => {
	// 	const saludo = 'Hola soy goku';
	// 	const { getByText } = render(<PrimeraApp saludo={saludo} />);
	//
	// 	expect(getByText(saludo)).toBeInTheDocument();
	// });
	test('Debe de mostrar <PrimeraApp /> correctamente', () => {
		const wrapper = shallow(<PrimeraApp saludo="Hola soy goku" />);

		expect(wrapper).toMatchSnapshot();
	});

	test('Debe de mostrar el subtitulo enviado por props', () => {
		const wrapper = shallow(
			<PrimeraApp saludo="Hola soy goku" subtitulo="Soy un subtitulo" />
		);

		// find = querySelector
		const textoParrafo = wrapper.find('p').text();
		expect(textoParrafo).toBe("Soy un subtitulo");
	});
});
