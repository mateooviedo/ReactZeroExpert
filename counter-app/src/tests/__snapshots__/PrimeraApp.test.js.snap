// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraAPP /> Debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
